### ISSUE TITLE
Catboost with pandas.Category kills kernel in jupyter lab

### ISSUE THREAD

**Issue by maxschatzki** [catboost/catboost#2122]
Problem: if features are as objects, then training takes forever and hangs at around 80% of the iterations specified. For example, if interation=10 it hangs at around 8th, if interation=50, it hangs at around 30-40 iterations and takes forever to finish. To speed up training object columns were converted to Category, but then as soon as training is started, kernel dies.
catboost version: 0.26.1
Operating System: Ubuntu 20.04
CPU: x
GPU:

Code. It is a binary classification problem with 5 catogorical features and around 9 numerical. Number of records = 2.6 mln.

scale_pos_weight = 1
classification_model = CatBoostClassifier(
    iterations=10,
    random_seed=63,
    learning_rate=0.7,
    custom_loss=['AUC', 'Accuracy'],
    scale_pos_weight = scale_pos_weight,
    boosting_type='Plain',
    thread_count=-1,
    depth=3,
)

classification_model.fit(
    X_train_cl_occurences, y_train_cl_occurences,
    cat_features=cat_features,
    eval_set=(X_train_cl_occurences, y_train_cl_occurences),
    verbose=True,
    plot=False,
)

**Comment by maxschatzki** [comment]
I do not know if it matters:

1 categorical feature - 59 unique values
2 - 7 unique values
3 - 3 unique values
4 - 6 unique values
5 - 6 unique values

I am not here to bash CatBoost, but after switching to LightGBM it trained on the same data in less than a minute.
